app/api/api_v1/endpoints/instructor_router.py

```diff
@@ -0,0 +1,49 @@
+from typing import List
+from pydantic import BaseModel
+from fastapi import APIRouter, Request, Depends
+from sqlalchemy.orm import Session
+from app.schemas import InstructorSchema
+from app.models.user import UserType
+from app.services import InstructorService, CourseService, KubernetesService
+from app.core.utils.auth_helper import PasswordHelper
+from app.core.dependencies import get_db, PermissionDependency, InstructorListPermission, InstructorCreatePermission
+
+router = APIRouter()
+
+class CreateInstructorWithoutPasswordBody(BaseModel):
+    onyen: str
+    first_name: str
+    last_name: str
+    email: str
+
+@router.get("/instructor/{onyen:str}", response_model=InstructorSchema)
+async def get_instructor(
+    *,
+    db: Session = Depends(get_db),
+    perm: None = Depends(PermissionDependency(InstructorListPermission)),
+    onyen: str
+):
+    instructor = await InstructorService(db).get_user_by_onyen(onyen)
+    return instructor
+
+@router.get("/instructors", response_model=List[InstructorSchema])
+async def list_instructor(
+    *,
+    request: Request,
+    db: Session = Depends(get_db),
+    perm: None = Depends(PermissionDependency(InstructorListPermission))
+):
+    instructors = await InstructorService(db).list_instructors()
+    return instructors
+
+@router.post("/instructor", response_model=InstructorSchema)
```

Comment:
Might be small, but I think we should use the plural as the root everywhere, e.g. `/instructors`. This also goes for line 19 as well as the student endpoints. When you're doing a POST on `/instructor` I don't think that makes much sense, since there's no resource for a single `instructor` without more information (like query params), only a list of instructors. And when you do a GET it should be `/instructors/{onyen:str}` because the instructor's onyen is a subset of the `instructors`. Only reason I bring it up is because this is when we're creating these endpoints, and if we create them this way it's much harder to change them later.

Reply:
@Hoid Right now, we use the plural of the resource name to refer to a collection of the resource. We use the singular to refer to an individual instance of the resource. There doesn't seem to be a completely agreed upon naming scheme for REST APIs. Some people like all resource names to be plural everywhere, some people like it how we have it right now.

Resource creation should definitely be plural, like POST /instructors. I'll fix this, it's a mistake. Do you want the API to be all plural?

Reply:
I think when you're dealing with lists of resources then yes, it should be all plural, but there are times when endpoints will be actions for the server to take rather than resources to return. In the case of an action it's likely not going to be plural. 

But for these endpoints I think endpoints underneath `/instructors` would pertain to those `instructors` (and the same goes for `/students`. It doesn't make sense to me that you'd create an instructor at 'POST /instructors` (which is right) and then go off of that subpath into a new path (`/instructor`) to query that same instructor you just created. I would think that all resources and actions that pertain to instructors would be underneath the `/instructors` path, so that we'd have nested paths, like how `/instructors` gives way to `/instructors/{onyen}`, and could potentially allow for something like `/instructors/{onyen}/profile` or `/instructors/{onyen}/class-rules` in the future.

Reply:
I'll fix the `post /instructors` typo and make sure that the rest of our endpoints are consistent with what you've described.